### PR TITLE
feat: Time-controlled status for Planned Maintenances / Schedules

### LINKED ISSUE #111: Time-controlled status for Planned Maintenances / Schedules
A practical extension for Planned Maintenances / Schedules would be the automatic time-controlled determination of the status.

| condition | status |
| - | - |
| now < scheduled_at | Upcoming |
| now > scheduled_at and now < completed_at | In Progress |
| now > completed_at | Complete |

This would be useful to announce maintenance carried out by third parties, where you only get information about the planned start and end.

I tried to hack this feature together, but I'm not versed enough with Laravel to implement this properly.

### PR DESCRIPTION
**Pull request by swoga**
Closes #111

- [x] drop `status` column
- [x] determine status from `scheduled_at` and `completed_at`
- [x] tests

### CODE REVIEW

File: database/migrations/2024_10_13_214300_drop_schedule_status.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('schedules', function (Blueprint $table) {
+            $table->dropColumn('status');
+            $table->timestamp('completed_at')->nullable(false)->change();
```
[the remark this answers is not in the document]
Comment (swoga) [reply database/migrations/2024_10_13_214300_drop_schedule_status.php]: Good point, I've changed it!

**Comment by jbrooksuk** [comment]
@swoga thanks! I've just pushed a small change to use `match` and the Carbon comparison methods. All we need now are some tests to ensure the status is correctly determined for each scenario.

**Comment by jbrooksuk** [comment]
Heh, Co pilot wrote the tests for me, I had to fix the factory. Thoughts?

**Comment by swoga** [comment]
> Heh, Co pilot wrote the tests for me, I had to fix the factory. Thoughts?

Nice, I've also started using Copilot in the last few days, it's helped me a lot with the previous PRs!

**Comment by jbrooksuk** [comment]
Nice!

Is there anything pending before we do a full review and merge?

**Comment by swoga** [comment]
I think the logic is good so far, only the tests need some work.
I've just quickly adapted the existing tests, so it might be necessary to rework them.
In particular, there should probably also be API tests with `completed_at` set to `null`.

**Comment by jbrooksuk** [comment]
@swoga great work on this. Are you able to finish the tests and we'll get this merged?
